### PR TITLE
feat: support ESLint 8.x

### LINKED ISSUE #218: Support ESLint 8.x
ESLint v8.0.0 is [released](https://eslint.org/blog/2021/10/eslint-v8.0.0-released) 🎉

It would be awesome to have official ESLint 8 support. 👊
I'm happy to help where I can of course 🙂

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
ESLint v8.0.0 is [released](https://eslint.org/blog/2021/10/eslint-v8.0.0-released) 🎉

devDependency compatibility with ESLint 8:

- [ ] [`eslint-config-prettier`](https://github.com/prettier/eslint-config-prettier) (https://github.com/prettier/eslint-config-prettier/issues/200)
  - [ ] https://github.com/prettier/eslint-config-prettier/pull/201
  - [x] ~Release~
- [x] [`eslint-plugin-eslint-plugin`](https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin) (https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/issues/175)
  - [x] https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/pull/119
  - [x] https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/pull/151
  - [x] https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/pull/186
  - [x] https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/pull/212
  - [x] [`v4.0.0-0`](https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/blob/master/CHANGELOG.md#v400-0)
  - [x] [`v4.0.0`](https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/releases/tag/v4.0.0)
- [x] [`eslint-plugin-jest`](https://github.com/jest-community/eslint-plugin-jest) (https://github.com/jest-community/eslint-plugin-jest/issues/881)
  - [x] https://github.com/jest-community/eslint-plugin-jest/pull/891
  - [x] https://github.com/jest-community/eslint-plugin-jest/pull/893
  - [x] https://github.com/jest-community/eslint-plugin-jest/pull/898
  - [x] https://github.com/jest-community/eslint-plugin-jest/pull/899
  - [x] https://github.com/jest-community/eslint-plugin-jest/pull/940
  - [x] https://github.com/jest-community/eslint-plugin-jest/pull/941
  - [x] https://github.com/jest-community/eslint-plugin-jest/pull/942
  - [x] [`v24.4.1`](https://github.com/jest-community/eslint-plugin-jest/releases/tag/v24.4.1)
  - [x] [`v24.4.2`](https://github.com/jest-community/eslint-plugin-jest/releases/tag/v24.4.2)
  - [x] [`v25.0.0`](https://github.com/jest-community/eslint-plugin-jest/releases/tag/v25.0.0)
  - [x] [`v25.0.4`](https://github.com/jest-community/eslint-plugin-jest/releases/tag/v25.0.4)
  - [x] [`v25.0.5`](https://github.com/jest-community/eslint-plugin-jest/releases/tag/v25.0.5)
  - [x] [`v25.1.0`](https://github.com/jest-community/eslint-plugin-jest/releases/tag/v25.1.0)
- [ ] [`eslint-plugin-node`](https://github.com/mysticatea/eslint-plugin-node) (https://github.com/mysticatea/eslint-plugin-node/issues/294)
  - [ ] https://github.com/mysticatea/eslint-plugin-node/pull/224
  - [ ] https://github.com/mysticatea/eslint-plugin-node/pull/295
  - [ ] Release
- [x] [`eslint-plugin-prettier`](https://github.com/prettier/eslint-plugin-prettier) (https://github.com/prettier/eslint-plugin-prettier/issues/427)
  - [x] https://github.com/prettier/eslint-plugin-prettier/pull/428
  - [x] ~Release~
- [x] [`jest-runner-eslint`](https://github.com/jest-community/jest-runner-eslint) (https://github.com/jest-community/jest-runner-eslint/issues/122)
  - [x] https://github.com/jest-community/jest-runner-eslint/pull/123
  - [x] https://github.com/jest-community/jest-runner-eslint/pull/128
  - [x] [`v0.11.0`](https://github.com/jest-community/jest-runner-eslint/releases/tag/v0.11.0)
  - [x] [`v1.0.0`](https://github.com/jest-community/jest-runner-eslint/releases/tag/v1.0.0)

---

Closes #218

### CODE REVIEW

**Comment by voxpelli** [comment]
Some changes needs to be made to the rules of this plugin:

* `eslint-plugin/require-meta-schema` doesn't pass on many of them
* `eslint-plugin/prefer-object-rule` doesn't pass on some of them

**Comment by RedGuy12** [comment]
why do dev dependencies need to be compatible? you can still use v7 for lining this, right?

**Comment by koooge** [comment]
Hi there, I need this peerDeps change to update eslint with npm@8.

eslint-plugin-node seems to support eslint 5+.  https://github.com/mysticatea/eslint-plugin-node/blob/master/package.json#L12-L14

So I suppose eslint-plugin-promise doesn't need to wait to update it. What is your concern about? @MichaelDeBoey

File: package.json
Comment (Belco90) [inline package.json:28]: Sticking to ESLint v7 in the dev dependencies would unblock this PR could since the main changes for adding ESLint v8 compatibility to the package are already addressed in this very PR. Upgrading ESLint here is just for the local environment used to develop the project, that must not block this PR to add ESLint v8 compatibility to the package.

**Review by Belco90 — CHANGES_REQUESTED** [review]

**Comment by niksy** [comment]
@xjamundx what can we do to help to get this merged?

**Comment by voxpelli** [comment]
> @xjamundx what can we do to help to get this merged?

I think we could break apart the needed changes into individual issues and PR:s which can more easily be reviewed and discussed rather than this large one which changes 18 files and has discussions all over the place.

Generally: The smaller change, the easier the merge and the more focused discussions and reviews.

**Comment by marcoreni** [comment]
> > @xjamundx what can we do to help to get this merged?
> 
> I think we could break apart the needed changes into individual issues and PR:s which can more easily be reviewed and discussed rather than this large one which changes 18 files and has discussions all over the place.
> 
> Generally: The smaller change, the easier the merge and the more focused discussions and reviews.

I generally agree on this, but 90% of the changed files only include `schema: [],`, so it should be pretty easy to review...

**Comment by voxpelli** [comment]
> I generally agree on this, but 90% of the changed files only include `schema: [],`, so it should be pretty easy to review...

Then make a PR for just those 90%, have it be easily review and merged on its own and not obscure the 10% that isn't that small and focused?

**Comment by iambumblehead** [comment]
eslint 8 was released 50 days ago, on Oct 9

**Comment by xjamundx** [comment]
So what's the status of this, can I help get it across?

**Comment by niksy** [comment]
@xjamundx I guess you could take a look at the changes in this PR and sign them off. @voxpelli mentioned separating some changes from this PR, but I guess you could decide if that’s needed?

**Comment by xjamundx** [comment]
@niksy I'll take a look and try to come up with a plan thanks

**Comment by MichaelDeBoey** [comment]
@voxpelli @niksy @xjamundx I've merged #227, #228, #229, #230 & #231 and added one last commit to support ESLint v8.x

**Review by Sandesh333333 — APPROVED** [review]

**Review by Sandesh333333 — APPROVED** [review]

**Comment by niksy** [comment]
@xjamundx I thinks PR’s that @MichaelDeBoey created are good enough to review as opposed to one large PR. This will make updating to latest ESLint easier.

**Comment by MichaelDeBoey** [comment]
Ping @xjamundx

**Comment by xjamundx** [comment]
okay let me take a look at everything this weekend. would love to get this published soon

**Comment by MichaelDeBoey** [comment]
@xjamundx if you need any help, you can always ping me 😉

**Comment by MichaelDeBoey** [comment]
@xjamundx I think this one can be merged now

**Comment by MichaelDeBoey** [comment]
@xjamundx Thanks for taking the time to merge this!

If you're publishing a new version, be sure to make it a major, as we've dropped Node 10

**Comment by xjamundx** [comment]
Give me a bit but yes we'll get it published soon. Fh

**Pull request merged**
